library/bootloader_facts.py

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2023, Sergei Petrosian <spetrosi@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: bootloader_facts
+
+short_description: Gather information for kernels as Ansible facts
+
+version_added: "0.0.1"
+
+description:
+    - "WARNING: Do not use this module directly! It is only for role internal use."
+    - Gather information for kernels as Ansible facts
+
+author:
+    - Sergei Petrosian (@spetrosi)
+"""
+
+EXAMPLES = r"""
+# Run with no parameters to gather facts
+bootloader_facts:
+"""
+
+import re
+from ansible.module_utils.basic import AnsibleModule
+
+
+def run_module():
+    # define available arguments/parameters a user can pass to the module
+    module_args = dict()
+
+    # seed the result dict in the object
+    # we primarily care about changed and state
+    # changed is if this module effectively modified the target
+    # state will include any data that you want your module to pass back
+    # for consumption, for example, in a subsequent task
+    result = dict(changed=False, ansible_facts=dict(bootloader_facts=dict()))
+
+    # the AnsibleModule object will be our abstraction working with Ansible
+    # this includes instantiation, a couple of common attr would be the
+    # args/params passed to the execution, as well as if the module
+    # supports check mode
+    module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+
+    # Keys from `grubby --info=ALL`, values for which this module returns
+    grubby_keys = ["args", "id", "index", "initrd", "root", "title"]
+
+    rc, stdout, stderr = module.run_command("which grubby", use_unsafe_shell=True)
+    if rc == 1:
+        module.fail_json(msg="grubby cli is not available", **result)
+
+    rc, stdout, stderr = module.run_command("grubby --info=ALL", use_unsafe_shell=True)
+    if "Permission denied" in stderr:
+        module.fail_json(msg="You must run this as sudo", **result)
+
+    kernels = re.findall('kernel="(.*)"', stdout)
+    for kernel in kernels:
+        if kernel not in result["ansible_facts"]["bootloader_facts"]:
+            result["ansible_facts"]["bootloader_facts"][kernel] = {}
+        rc, stdout, stderr = module.run_command(
+            "grubby --info=%s" % kernel, use_unsafe_shell=True
+        )
```

Comment:
Why do you have to call `grubby` again?  Doesn't `grubby --info=ALL` return all possible data for all kernels?

Reply:
`grubby --info=ALL` returns kernel info without formatting, I found it easier to call grubby for each kernel. I can see that running multiple commands when all the info is available already is bad, I'll try to process `--info=ALL` output.

Reply:
How does the output of `grubby --info=ALL` differ from `grubby --info=$kernel`?  It looks the same on my f38 system.

Reply:
Fixed